src/python_pybind11/src/Vector2.hh

```diff
@@ -0,0 +1,139 @@
+                    pyclass_name.c_str(),
+                    py::buffer_protocol(),
+                    py::dynamic_attr())
+    .def(py::init<>())
+    .def(py::init<const T&, const T&>())
+    .def(py::init<const Class>())
+    .def("sum", &Class::Sum, "Return the sum of the values")
+    .def("distance", &Class::Distance, "Calc distance to the given point")
+    .def("length",
+         &Class::Length,
+         "Returns the length (magnitude) of the vector")
+    .def("squared_length",
+         &Class::SquaredLength,
+         "Return the square of the length (magnitude) of the vector")
+    .def("normalize", &Class::Normalize, "Normalize the vector length")
+    .def("normalized", &Class::Normalized, "Return a normalized vector")
+    .def("round", &Class::Round)
+    .def("rounded", &Class::Rounded, "Get a rounded version of this vector")
+    .def("set", &Class::Set, "Set the contents of the vector")
+    .def("dot",
+         &Class::Dot,
+         "Return the dot product of this vector and another vector")
+    .def("abs", &Class::Abs,
+        "Return the absolute dot product of this vector and "
+        "another vector. This is similar to the Dot function, except the "
+        "absolute value of each component of the vector is used.")
+    .def("abs_dot", &Class::AbsDot, " Get the absolute value of the vector")
+    .def("max",
+         py::overload_cast<const Class&>(&Class::Max),
+         "Set this vector's components to the maximum of itself and the "
+         "passed in vector")
+    .def("max", py::overload_cast<>(&Class::Max, py::const_),
+         "Get the maximum value in the vector")
+    .def("min", py::overload_cast<const Class&>(&Class::Min),
+         "Set this vector's components to the minimum of itself and the "
+         "passed in vector")
+    .def("min", py::overload_cast<>(&Class::Min, py::const_),
+         "Get the minimum value in the vector")
+    .def(py::self + py::self)
+    .def(py::self += py::self)
+    .def(py::self + T())
+    .def(py::self += T())
+    .def(py::self * py::self)
+    .def(py::self *= py::self)
+    .def(py::self * T())
+    .def(py::self *= T())
+    .def(py::self - py::self)
+    .def(py::self -= py::self)
+    .def(py::self - T())
+    .def(py::self -= T())
+    .def(py::self / py::self)
+    .def(py::self /= py::self)
+    .def(py::self / T())
+    .def(py::self /= T())
+    .def(py::self != py::self)
+    .def(py::self == py::self)
+    .def(-py::self)
+    .def("equal", &Class::Equal, "Equal to operator")
+    .def("is_finite",
+         &Class::IsFinite,
+         "See if a point is finite (e.g., not nan)")
+    .def("correct", &Class::Correct, "Corrects any nan values")
+    .def("x", py::overload_cast<>(&Class::X), "Get the x value.")
+    .def("y", py::overload_cast<>(&Class::Y), "Get the y value.")
+    .def("x", py::overload_cast<const T&>(&Class::X), "Get the x value.")
+    .def("y", py::overload_cast<const T&>(&Class::Y), "Get the y value.")
```

Comment:
I think this and the one above are setters, not getters. Please adjust the doc-string